cpu/esp32/irq_arch.c

```diff
@@ -19,58 +19,101 @@
  */
 
 #include "irq_arch.h"
+
+#include "esp_attr.h"
 #include "esp_err.h"
-#include "esp32/rom/ets_sys.h"
-#include "soc/dport_reg.h"
-#include "xtensa/xtensa_api.h"
+#include "freertos/FreeRTOS.h"
+#include "hal/interrupt_controller_types.h"
+#include "hal/interrupt_controller_ll.h"
+#include "rom/ets_sys.h"
+#include "soc/periph_defs.h"
+#include "esp_intr_alloc.h"
 
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-struct _irq_alloc_table_t {
+#ifndef ETS_CAN_INTR_SOURCE
+#define ETS_CAN_INTR_SOURCE     ETS_TWAI_INTR_SOURCE
+#endif
+
+typedef struct intr_handle_data_t {
     int      src;   /* peripheral interrupt source */
     uint32_t intr;  /* interrupt number */
```

Comment:
Does this need to be 32 bit?

Reply:
Good question. All functions of ESP-IDF use `int` for interrupt numbers which is 32 bit. Implicit type casts shouldn't be a problem but accessing smaller elements in structs might be less efficient.